src/Bicep.Core/TypeSystem/DeployTimeConstantValidator.cs

```diff
@@ -8,31 +8,107 @@
 using Bicep.Core.Extensions;
 using Bicep.Core.Semantics;
 using Bicep.Core.Syntax;
+using Bicep.Core.TypeSystem.Az;
 
 namespace Bicep.Core.TypeSystem
 {
     public static class DeployTimeConstantValidator
     {
         public static void Validate(SemanticModel semanticModel, IDiagnosticWriter diagnosticWriter)
         {
-            // Collect all sytnaxes that require DTCs (a.k.a. DTC containers).
-            var containers = DeployTimeConstantContainerVisitor.CollectDeployTimeConstantContainers(semanticModel);
+            var existingResourceBodyObjectTypeOverrides = new Dictionary<DeclaredSymbol, ObjectType>();
+
+            // grab all existing resources
+            var existingResourceSymbols = semanticModel.DeclaredResources
+                .Select(x => x.Symbol)
+                .Where(x => x.DeclaringResource.IsExistingResource())
+                .ToArray();
+
+            // compare existing resources to all other existing resources to find dependencies and see if it has a non-DTC value
+            // and needs to remove the ReadableAtDeployTime flag from the "name" property -> O(n^2) time complexity
+            for (int i = 0; i < existingResourceSymbols.Length; i++)
+            {
+                foreach (var existingResourceSymbol in existingResourceSymbols)
+                {
+                    // grab the "name" property to check if it has a DTC value or not
+                    var nameObjectPropertySyntax = existingResourceSymbol.DeclaringResource.TryGetBody()?.TryGetPropertyByName(AzResourceTypeProvider.ResourceNamePropertyName);
 
+                    var diagnosticWriterForExistingResources = ToListDiagnosticWriter.Create();
+
+                    // if "name" property has a non-DTC value, then make and entry for the corresponding existing ResourceSymbol to a modified ObjectType in the dictionary
+                    if (nameObjectPropertySyntax != null && !ContainsDeployTimeConstant(nameObjectPropertySyntax, nameObjectPropertySyntax.Value, semanticModel, diagnosticWriterForExistingResources, existingResourceBodyObjectTypeOverrides))
```

Comment:
I'd suggest changing `bool ContainsDeployTimeConstant` to `void CheckDeployTimeConstantViolations` and check if there are diagnostics collected by `diagnosticWriterForExistingResources` in the if condition.